ucrop/src/main/java/com/yalantis/ucrop/util/BitmapLoadUtils.java

```diff
@@ -53,53 +60,94 @@ public BitmapWorkerResult(@Nullable Bitmap bitmapResult, @Nullable Exception bit
     }
 
     /**
-     * Creates and returns a Bitmap for a given Uri.
+     * Creates and returns a Bitmap for a given Uri(String url).
      * inSampleSize is calculated based on requiredWidth property. However can be adjusted if OOM occurs.
      * If any EXIF config is found - bitmap is transformed properly.
      */
     static class BitmapWorkerTask extends AsyncTask<Void, Void, BitmapWorkerResult> {
 
         private final Context mContext;
         private final Uri mUri;
+        private final String mSource;
         private final int mRequiredWidth;
         private final int mRequiredHeight;
+        private final Boolean mDownloadable;
 
         private final BitmapLoadCallback mBitmapLoadCallback;
 
-        public BitmapWorkerTask(@NonNull Context context, @Nullable Uri uri,
+        public BitmapWorkerTask(@NonNull Context context,
+                                @Nullable Uri uri, @Nullable String source,
                                 int requiredWidth, int requiredHeight,
                                 BitmapLoadCallback loadCallback) {
             mContext = context;
             mUri = uri;
+            mSource = source;
             mRequiredWidth = requiredWidth;
             mRequiredHeight = requiredHeight;
             mBitmapLoadCallback = loadCallback;
+            mDownloadable = mUri == null;
+        }
+
+        static OkHttpClient client = new OkHttpClient();
+        static ResponseBody connect = null;
+        //  opens connection for a given string url
+        private ResponseBody connect(String url) throws IOException {
+            Request request = new Request.Builder()
+                    .url(url)
+                    .build();
+            return client.newCall(request).execute().body();
         }
 
         @Override
         @NonNull
         protected BitmapWorkerResult doInBackground(Void... params) {
-            if (mUri == null) {
```

Comment:
what you should do here:

1. Check Uri scheme (http,https,file or content).
2. If it's file/content - cool, logic already works with those.
3. If it's http/https - download image (not into Bitmap and then compress to file, but directly to file).
When it's downloaded - get it's Uri and go to step 2. (that's it)